docs/contributing-devguide.md

```diff
@@ -13,8 +13,16 @@ If you want to run controller tests you also need to install kubebuilder's `envt
 ```
 export KUBEBUILDER_TOOLS_VERSION='1.20.2' # check for latest version or a version that has support to what you are testing
```

Comment:
Just as a sidenote: [setup-envtest](https://github.com/kubernetes-sigs/controller-runtime/tree/master/tools/setup-envtest) can do the heavy lifting for us, this would simplify the documentation here a lot: we just `go install`, use it and point to the documentation of `setup-envtest`.

Reply:
Should we say to the person to set XDG_DATA_HOME and then just use envtest? Or just point it there directly?

Reply:
Just show the usage and point to the readme of the tool:

```bash
go install sigs.k8s.io/controller-runtime/tools/setup-envtest@latest

# list available versions
setup-envtest list --os $(go env GOOS) --arch $(go env GOARCH)

source <(setup-envtest use 1.21.x -p env --os $(go env GOOS) --arch $(go env GOARCH))

# for more information see here: 
# https://github.com/kubernetes-sigs/controller-runtime/tree/master/tools/setup-envtest
setup-envtest --help

```

Reply:
Yeah, I would like something very copy pastable with not much reading needed for users from our side. If we don't set $XDG_DATA_HOME it seems setup-envtest spits binaries in `~/.local/share/kubebuilder-envtest/k8s/VERSION` which is fine, but our tests expect things to be at `/usr/local/kubebuilder`.

Or we should be looking at `~/.local/share/kubebuilder-envtest/k8s/VERSION`, or letting users know that they should set $XDG_DATA_HOME right away, I think.

Reply:
The envtest package reads binaries from the `$KUBEBUILDER_ASSETS` env var. The above command `source <(...)` sets these variables. There should be nothing left to do. (the ci pipeline reads it from that dir which is the default, we gotta change that, too - but that's a different topic)

Reply:
Cool, you are right, Just tested it. Removed `/usr/local/kubebuilder`, installed `setup-envtest`, sourced the env vars, and got tests to work.

Only thing is that it is failing with ` timeout waiting for process kube-apiserver to stop` with 1.21.x. Had to use 1.20.2.

So, 
```
setup-envtest use -p path 1.20.2
source <(setup-envtest use 1.20.2 -p env --os $(go env GOOS) --arch $(go env GOARCH))
make test
```

Reply:
that kube-apiserver issue is documented here: https://github.com/kubernetes-sigs/controller-runtime/issues/1571
I'll provide fix for that